### PR TITLE
Add backend stuff for feature 'setting the page owner'

### PR DESCRIPTION
**Pull request by otofune**
_import backend stuff for feature 'setting page owner' from working branch from #371_

functions will be called with current #378 client
---
+ @api {post} /page_owners.activate Activate page owner configuration
+ @api {post} /page_owners.deactivate Deactivate page owner configuration
+ @api {get} /teams.list List all team
+ @api {post} /teams.create Create team
+ Add `bodyConfirmed` field to edit form

functions will not be called with current #378 client
---
+ @api {post} /pages.updateLifetime Set lifetime to page
+ @api {get} /teams.get Get team with given ID
+ @api {post} /teams.edit Edit team

### CODE REVIEW

File: lib/models/page.js [hunk 1]
```diff
@@ -885,8 +960,10 @@ module.exports = function(crowi) {
     const Revision = crowi.model('Revision')
     const Bookmark = crowi.model('Bookmark')
     const grant = options.grant || null
+
+    const { expirationAt = null } = pageData.populated('revision') ? pageData.revision : null
```
Comment (otofune) [inline lib/models/page.js]: 👀!!!!

Comment (otofune) [reply lib/models/page.js]: I'll explain more detail & add endpoint to acknowledge the document as correct

File: lib/models/page.js [hunk 2]
```diff
@@ -830,84 +906,122 @@ module.exports = function(crowi) {
...
+        pageEvent.emit('update', p, user, bookmarkCount)
+        return p
+      }
+
+      pageEvent.emit('update', savedPage, user, bookmarkCount)
+      return savedPage
+    } catch (err) {
+      debug('Error on update', err)
+      debug('Error on update', err.stack)
+      throw err
```
Comment (otofune) [inline lib/models/page.js]: In previous impl, users can't notice errors that occured at updating. So, I change behavior on this methods at first step.

File: lib/models/page.js [hunk 3]
```diff
@@ -830,84 +906,122 @@ module.exports = function(crowi) {
...
+      const bookmarkCount = await Bookmark.countByPageId(pageData._id)
+
+      // If need to update grant
+      if (grant != pageData.grant) {
+        const p = await Page.updateGrant(savedPage, grant, user)
+        pageEvent.emit('update', p, user, bookmarkCount)
+        return p
+      }
+
+      pageEvent.emit('update', savedPage, user, bookmarkCount)
```
Comment (otofune) [inline lib/models/page.js]: In previous impl, this method emit pageData to 'update' event and return pageData (argument). I think it's better to use new one, so changed.

File: lib/routes/team.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+   * @apiParam {String} name?
+   */
+  api.edit = wrap(async (req, res) => {
+    const { team_id: teamId, name } = req.body
+
+    const team = await Team.findById(teamId)
+    if (!team) return res.status(404).json(ApiResponse.error('There are no team with given id.'))
+
+    team.name = name
+    res.json(ApiResponse.success())
```
[the remark this answers is not in the document]
Comment (otofune) [reply lib/routes/team.js]: Thank you for your good catch!!! Yes I forgot... I'll move this logic to team model and write test

File: lib/routes/team.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    if (users.length !== userids.length) return res.status(400).json(ApiResponse.error('There are missing users with given id.'))
+
+    await team.deleteUser(...users)
+    return res.json(ApiResponse.success())
+  })
+
+  const middleware = {}
+
+  middleware.handleError = (e, _, res) => {
+    debug(e)
```
[the remark this answers is not in the document]
Comment (otofune) [reply lib/routes/team.js]: Thank you! I'll move this to under L151

File: lib/models/revision.js [hunk 6]
```diff
@@ -10,6 +10,7 @@ module.exports = function(crowi) {
     format: { type: String, default: 'markdown' },
     author: { type: ObjectId, ref: 'User' },
     createdAt: { type: Date, default: Date.now },
+    expirationAt: Date,
```
[the remark this answers is not in the document]
Comment (otofune) [reply lib/models/revision.js]: I see, I'll use 'expireAt'!

**Comment by otofune** [comment]
@sotarok I've fixed! Could you review again?
I've removed `addUser` & `deleteUser`. There are replaced by `edit`.

File: test/models/team.test.js [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+      }),
+    ])
+  })
+
+  /**
+   * class methods
+   */
+
+  describe('#findByUser', () => {
+    it('Find teams by user collectly', async () => {
```
Comment (lightnet328) [inline test/models/team.test.js]: `correctly` is probably correct.

File: test/models/team.test.js [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+      await createTeam(...createdUsers)
+      const team0 = await createTeam(createdUsers[0])
+      const team1 = await createTeam(createdUsers[1])
+
+      const teamsRelatedTo0 = await Team.findByUser(createdUsers[0])
+      expect(teamsRelatedTo0).toHaveLength(2)
+      // ここらへんの assert うまいことできんかな
+      expect(teamsRelatedTo0.map(team => team._id.toString())).toEqual(expect.not.arrayContaining([team1._id.toString()]))
```
Comment (lightnet328) [inline test/models/team.test.js]: If writing shorter, you can write followings.
```js
expect(teamsRelatedTo0.every(team => !team._id.equals(team1._id))).toBe(true)
```
However, I'd rather use `.arrayContaining` than `.toBe`.

Comment (otofune) [reply test/models/team.test.js]: Ya, we can use custom expect with https://github.com/facebook/jest/issues/256 workaround, but ... not clever.... So I uses your shorter suggestion!

**Comment by otofune** [comment]
I'll recreate if requires this feature.

**Pull request closed without merging**